sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -156,9 +156,16 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       }
       WriteToDataSourceV2Exec(writer, invalidateCacheFunc, planLater(query), customMetrics) :: Nil
 
-    case CreateV2Table(catalog, ident, schema, parts, props, ifNotExists) =>
+    case CreateTable(ResolvedDBObjectName(catalog, ident), schema, partitioning, bucketSpec,
+        tableSpec, ifNotExists) =>
+      val props = CatalogV2Util.convertTableProperties(
```

Comment:
I think it's better to keep the strategy rule simple and move the logic of generating table properties into `CreateTableExec`